src/main/scala/gen/gmos/GmosNLongslitD.scala

```diff
@@ -0,0 +1,250 @@
+// Copyright (c) 2019 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package basic
+package gen.gmos
+
+import basic.search.Constraints
+import basic.syntax.all._
+
+import gem.Step
+import gem.config.DynamicConfig.GmosN
+import gem.config.GmosConfig._
+import gem.enum._
+import gem.math.{ MagnitudeValue, Wavelength }
+
+import cats.Functor
+import cats.effect.Sync
+import cats.implicits._
+import fs2.Stream
+import monocle.{Prism, Optional}
+
+import java.time.Duration
+import java.util.concurrent.TimeUnit.NANOSECONDS
+
+import scala.concurrent.duration.FiniteDuration
+
+/**
+ * Dynamic sequence generation for GMOS North Longslit.
+ */
+sealed trait GmosNLongslitD {
```

Comment:
Could we have `GmosNLongslitD[F[_]]` and remove the `Sync` constraint on `sequence`? We can demand it instead on `GmosNLongslitD.apply`. This would be a more idiomatic encoding.

Reply:
Yes, that seems better.